module2/module2.js

```diff
@@ -0,0 +1,196 @@
+// ==== Module#2 Functions ====
+mocha.setup('bdd');
+const expect = chai.expect;
+
+// ==== Currying ====
+
+function curry(f) {
+  return function(a) {
+    return function(b) {
+      return function(c) {
+        return function(d) {
+          return f(a, b, c, d);
+        };
+      };
+    };
+  };
+};
+
+function joinFunction(...args) {
+  return args.join(' ');
+}
+
+const mergeWords =  curry(joinFunction);
+
+describe('mergeWords', () => {
+  it('should merge words into sentence', () => {
+    const result = mergeWords('GNU')('is')('not')('Unix.'); //removed function call, is it ok?
+
+    expect(result).to.equal('GNU is not Unix.');
+  });  
+});
+
+// ==== Every/Some ====
+const goodUsers = [
+    { id: 1 },
+    { id: 2 },
+    { id: 3 },
+];
+
+// `checkUsersValid` is the function you'll define
+const checkUsersValid = (goodUsers) => {
+  return function(usersToTest) {
+    let absentUsers = usersToTest.filter( (user) => filterAbsentUser(user, goodUsers))
+    return absentUsers.length === 0; 
+  };
+};
+
+function filterAbsentUser(user, goodUsers) {
```

Comment:
I would remove filterAbsentUser function and create more clean and clear solution.
You have goodUsers argument is saved in the Lexical Environment, you have usersToTest.
Both are arrays. Just find one array items on another one array in a loop using [].find